internal/bminventory/inventory.go

```diff
@@ -4774,9 +4774,22 @@ func (b *bareMetalInventory) validateInfraEnvCreateParams(ctx context.Context, p
 	}
 
 	if params.InfraenvCreateParams.StaticNetworkConfig != nil {
-		if err = b.staticNetworkConfig.ValidateStaticConfigParams(params.InfraenvCreateParams.StaticNetworkConfig); err != nil {
+		// backward compatibility - nmstate.service has been available on RHCOS since version 4.14+, therefore, we should maintain both flows
+		var ok bool
+		ok, err = staticnetworkconfig.NMStatectlServiceSupported(params.InfraenvCreateParams.OpenshiftVersion, common.X86CPUArchitecture)
+		if err != nil {
 			return err
 		}
+
+		if ok {
```

Comment:
I agree. I'll keep the validation directly on the YAML since we don't want our code handling both YAML and nmconnection formats. The whole point of using nmstate is that we don't need to worry about its internal translation. Right now, we're using the generator for validation, but we're aware that the nmstate team is working on Go structures and adding validation, so that aspect will change in a later phase.